[cpu] Support tl.load(..., padding="nan")

int3 opened the pull request:
This makes test_block_pointer.py pass.

Fixes #60.

int3 commented on third_party/cpu/lib/TritonToTritonCPU/ConvertMemoryOps.cpp:
```diff
@@ -126,8 +150,10 @@ struct LoadOpConversion : public MemoryOpConversion<triton::LoadOp> {
     }
-    auto vecRead = rewriter.create<vector::TransferReadOp>(loc, resTy, memRef,
-                                                           indices, inBounds);
+    Value padding = getPaddingValue(loc, resTy.getElementType(),
```
I don't think it matters whether I get the type from the result type or the input type, but not 100% sure...

ienkovich replied:
The result type usage seems right.

int3 commented on third_party/cpu/lib/TritonToTritonCPU/ConvertMemoryOps.cpp:
```diff
@@ -96,6 +96,30 @@ struct MemoryOpConversion : public OpConversionPattern<OpT> {
+      break;
+    case triton::PaddingOption::PAD_NAN:
+      assert(!type.isIntOrIndex());
+      auto apNaN = llvm::APFloat::getNaN(
+          FloatAttr::get(type, 0).getValue().getSemantics());
```
this seems like a rather awkward way to get the NaN value, but it's more or less what RewriteTensorPointer.cpp does

ienkovich replied:
Yeah, there is no good helper in rewriter to build a NaN. For the semantics, I'd use `cast<FloatType>(type).getFloatSemantics()`.

ienkovich commented on third_party/cpu/lib/TritonToTritonCPU/ConvertMemoryOps.cpp:
```diff
@@ -96,6 +96,30 @@ struct MemoryOpConversion : public OpConversionPattern<OpT> {
+    TypedAttr attr;
+    switch (padding.value()) {
+    case triton::PaddingOption::PAD_ZERO:
+      attr = type.isIntOrIndex() ? cast<TypedAttr>(IntegerAttr::get(type, 0))
```
You can use `rewriter.getZeroAttr()` here

ienkovich reviewed (approved):
Looks good, thanks!

int3 commented:
@minjang I haven't addressed comments yet 😅 also I introduced a small bug which I want to fix... will put up a separate PR I guess

Since I have merge permissions now, could you let me do the merging for my own PRs next time? Thanks!

minjang commented:
> @minjang I haven't addressed comments yet 😅 also I introduced a small bug which I want to fix... will put up a separate PR I guess
> 
> Since I have merge permissions now, could you let me do the merging for my own PRs next time? Thanks!

Lol. Sorry for that. Yes, I will leave the PRs later on!
